src/core/components/collapsible/collapsible.tsx

```diff
@@ -0,0 +1,75 @@
+import classNames from 'classnames';
+import { useCallback, useState } from 'react';
+import { Button } from '../button';
+import { Icon, IconType } from '../icon';
+import { type ICollapsibleProps } from './collapsible.api';
+
+const collapsedSizeVariants = {
+    sm: 'h-32',
+    md: 'h-64',
+    lg: 'h-96',
+};
+
+export const Collapsible: React.FC<ICollapsibleProps> = ({
+    collapsedSize = 'md',
+    customCollapsedSize,
+    defaultOpen = false,
+    buttonLabelOpened,
+    buttonLabelClosed,
+    buttonVariant,
+    className,
+    blinderClassName,
+    onToggle,
+    children,
+    ...otherProps
+}) => {
+    const [isOpen, setIsOpen] = useState(defaultOpen);
+
+    const toggle = useCallback(() => {
+        setIsOpen((prev) => {
+            const newState = !prev;
+            onToggle?.(newState);
+            return newState;
+        });
+    }, [onToggle]);
+
+    const contentClassNames = classNames(
+        'mb-4 overflow-hidden ease-in-out md:mb-6',
+        isOpen ? 'h-auto' : collapsedSizeVariants[collapsedSize],
+    );
+
+    return (
+        <div className={className} {...otherProps}>
+            <div
+                style={{ height: !isOpen && customCollapsedSize ? `${customCollapsedSize}px` : undefined }}
```

Comment:
Referencing the api comment..If the `customCollapsedSize` is of type `CSSProperties['height']`, we can then simplify this to something like the following:
```jsx
<... style={{height: !isOpen && customCollapsedSize}}>
```